kornia/augmentation/functional.py

```diff
@@ -396,8 +396,10 @@ def apply_perspective(input: torch.Tensor, params: Dict[str, torch.Tensor],
     mask = params['batch_prob'].to(input.device)
 
     # TODO: look for a workaround for this hack. In CUDA it fails when no elements found.
+    # TODO: this if statement is super weird and sum here is not the propeer way to check
+    # it's valid. In addition, 'interpolation' shouldn't be a reason to get into the branch.
 
-    if bool(mask.sum() > 0):
+    if bool(mask.sum() > 0) and ('interpolation' in params):
```

Comment:
@shijianjian check this. Just workaround. Consider merging this before your decoupling PR